pkg/storage/test_date.go

```diff
@@ -109,3 +109,30 @@ func (s *Store) UpdateTestDate(ctx context.Context, testDate *models.TestDate) (
 	}
 	return s.FetchTestDateByID(ctx, testDate.ID)
 }
+
+// RemoveTestDate removes (soft delete) an existing test date object in the database
+func (s *Store) RemoveTestDate(ctx context.Context, testDate *models.TestDate) (*models.TestDate, error) {
+	updatedAt := s.clock.Now()
```

Comment:
This time should be in UTC

Reply:
I imagine we want to change this in CreateTestDate and UpdateTestDate as well?